src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -449,7 +449,7 @@ internal static bool IsPackageReferenceProject(Project project)
                     }
                     // If no matching packages were found, then the package is transitive,
                     // and include-transitive must be used to add the package
-                    else if (transitive)
+                    else if (transitive && (library.Type == "package")) // be sure to exclude "project" references here as these are irrelevant
```

Comment:
You can remove the parenthesis

```suggestion
                    else if (transitive && library.Type == "package") // be sure to exclude "project" references here as these are irrelevant

```